attest/verifier/types/src/convert.rs

```diff
@@ -16,6 +18,23 @@ pub enum ConversionError {
     InvalidContents(String),
     /// Other error: {0}
     Other(String),
+    /**
+     * The length of `{name}` does not match the expected
+     * length, provided {provided}, required {required}
+     */
+    LengthMismatch {
+        name: String,
+        provided: usize,
+        required: usize,
+    },
+    /// The key is not valid: {0}
+    Key(KeyError),
```

Comment:
I brought in the KeyError as that's what `mc-api` uses in it's `ConversionError` and it might be good to be able to map as many as possible, more easily.